pkg/engine/executor/dataobjscan.go

```diff
@@ -67,6 +68,9 @@ func (s *dataobjScan) Read() error {
 	s.state = newState(rec, err)
 
 	if err != nil {
+		if err == EOF {
+			return err
+		}
```

Comment:
Needing to do this makes me feel like we're missing a `errors.Is(err, EOF)` somewhere and doing a `err == EOF` instead; ideally we shouldn't have to unwrap errors like this